relbench/external/graph.py

```diff
@@ -215,6 +218,7 @@ def get_link_train_table_input(
     table: Table,
     task: LinkTask,
 ) -> LinkTrainTableInput:
+    table.df = table.df.reset_index()
```

Comment:
I previously added [this line](https://github.com/snap-stanford/relbench/blob/main/relbench/data/task_link.py#L67) to address the segfault issue.

Can you clarify why this is not enough?

Reply:
There's no dangling foreign keys. However, I dropped the nan's in the train table. So there are 7614 rows, but the max index is 7631.
```
(Pdb) filter_mask.any()
False
(Pdb) c
> /home/ubuntu/git/relbench/relbench/external/graph.py(221)get_link_train_table_input()
-> table.df = table.df.reset_index()
(Pdb) table.df
      timestamp   user                                              event
0    2012-06-17   1049                                           [276785]
1    2012-07-07    269           [362015, 365602, 365516, 360563, 371187]
2    2012-07-07   4284                           [365835, 365590, 391659]
3    2012-07-07   6723  [389531, 391659, 371821, 394730, 361664, 37297...
4    2012-07-07   1816                                           [396226]
...         ...    ...                                                ...
7627 2012-09-15   1567                                           [877156]
7628 2012-09-15   5193                                           [886905]
7629 2012-09-15   4142                                           [912626]
7630 2012-09-15  14332                                           [977860]
7631 2012-09-15   5603                                           [979524]

[7614 rows x 3 columns]
```

Reply:
Do you know why there is nan in train table?

Reply:
I thought dangling fkey is the only source, but I want to get some clarification why this could happen.

Reply:
In my case it's the link prediction train table contains nans(after performing `duckdb.sql()`), so I filtered out the nan's but didn't reset index. The table I have looks like the following:

```
      timestamp     user                                              event
57   2012-09-15      NaN  [852657, 854664, 854670, 855787, 856058, 85869...
58   2012-09-15  11493.0  [859207, 904469, 976295, 926508, 947218, 97700...
59   2012-09-15   4318.0                                           [858343]
60   2012-09-15   5217.0           [858425, 977616, 911618, 993568, 977000]
61   2012-09-15  10972.0  [878296, 976175, 994197, 947218, 912626, 94659...
...         ...      ...                                                ...
7592 2012-09-15  13207.0                                           [871611]
7593 2012-09-15    177.0                                           [882423]
7594 2012-09-15   6805.0                                           [968540]
7595 2012-09-15  12523.0                                           [977860]
7596 2012-09-15   7098.0                                           [977860]
```
So after i filter out nan, index `57` won't exist. So the indices in the train table could be larger than the size of the train table and thus causing seg fault.

Reply:
`get_link_train_table_input` will break if the table df contains nan values in src col. So users will have to filter out the nan's themselves.

Reply:
Why don't you write `duckdb.sql` such that you don't have NaN user?

Reply:
My suggestion would be to understand why `duckdb.sql` produces nan. Once we understand it, we either modify SQL or just add something like https://github.com/snap-stanford/relbench/blob/main/relbench/tasks/stackex.py#L208-L209

Reply:
we don't want to have NaN in our train table.

Reply:
Sure I will modify that in the task and remove changes here.